lib/route_patterns/repo.ex

```diff
@@ -38,16 +45,17 @@ defmodule RoutePatterns.Repo do
     opts
     |> Keyword.put(:route, route_id)
     |> Keyword.put(:sort, "typicality,sort_order")
-    |> cache(&api_all/1)
+    |> api_all()
     |> Enum.sort(&reorder_mrts(&1, &2, route_id))
   end
 
   def by_stop_id(stop_id) do
     [stop: stop_id]
     |> Keyword.put(:include, "representative_trip.shape,representative_trip.stops")
-    |> cache(&api_all/1)
+    |> api_all()
   end
 
+  @decorate cacheable(cache: @cache, on_error: :nothing, opts: [ttl: @ttl])
   defp api_all(opts) do
```

Comment:
The example above I was just re-forming the logic to fit in with the decorator pattern. With `api_all` it's a lot simpler. But, I wasn't trying to change the logic because I don't really know if a consumer wants or expects an empty array here. I can see cases where you would want to cache an empty list if you don't expect that it won't be empty next time you request it.